### PR TITLE
[WIP] Add initial support for IMX8

### PR DESCRIPTION
**Pull request by dianacretu**

### CODE REVIEW

File: include/hw/adsp/imx8.h [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+#include <sys/time.h>
+#include <stdio.h>
+#include <stdint.h>
+#include "qemu/osdep.h"
+#include "qapi/error.h"
+#include "qemu-common.h"
+#include "exec/hwaddr.h"
+#include "hw.h"
+
+/* Baytrail, Cherrytrail and Braswell */
```
Comment (lgirdwood) [inline include/hw/adsp/imx8.h:33]: comments for imx ?

File: target/xtensa/core-imx8.c [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2015, Intel Corporation.
```
Comment (lgirdwood) [inline target/xtensa/core-imx8.c:2]: copyright NXP ?

**Review by lgirdwood — CHANGES_REQUESTED** [review]
Can you also do this for sof-v4.2 - I guess @xiulipan will move CI over to v4.2 after the Chinese New year at some point so we should be ready for this and for upstreaming.

**Comment by lgirdwood** [comment]
@dianacretu please also add what, why and how comments to commit messages and PR descriptions.

**Review by lgirdwood — COMMENTED** [review]
@dianacretu any update on using the sof 4.2 branch, will be moving docker and CI over to this soon so it's best target for development.

**Review by lgirdwood — COMMENTED** [review]
@dianacretu are things fully working now ?

**Comment by dianacretu** [comment]
@lgirdwood booting works fine, thanks for review. Will rebase on sof 4.2 branch with your comments addressed. We need to do some cleanups, please DO NOT merge yet.

**Comment by lgirdwood** [comment]
@dianacretu ok, great - good to know its all booting now :)

**Comment by dianacretu** [comment]
Work done. Another pull request was created: https://github.com/thesofproject/qemu/pull/30

**Pull request closed without merging**